tests/e2e/config_secret_utils.go

```diff
@@ -0,0 +1,563 @@
+		hostsResult, err := sshExec(sshClientConfig, masterIp, hosts)
+		if err != nil && hostsResult.Code != 0 {
+			fssh.LogResult(hostsResult)
+			return nil, fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				hosts, masterIp, err)
+		}
+		if hostsResult.Stdout != "" {
+			hostListTemp := strings.Split(hostsResult.Stdout, "\n")
+			hostList = append(hostList, hostListTemp...)
+		}
+		for i := 0; i < len(hostList); i++ {
+			if hostList[i] != "" {
+				hostsList = append(hostsList, hostList[i])
+			}
+		}
+		hostList = nil
+	}
+	return hostsList, nil
+}
+
+// getVmDetails method is used to fetch vm details
+func getVmDetails(vcAddress string, vCenterPort string, sshClientConfig *ssh.ClientConfig,
+	masterIp string, dataCenter []string) ([]string, error) {
+	var vmsList []string
+	var vmList []string
+	// get vm details
+	ginkgo.By("Executing command for fetching k8s vm details")
+	for i := 0; i < len(dataCenter); i++ {
+		vms := govcLoginCmd(vcAddress, vCenterPort) + "govc ls /" + dataCenter[i] + "/vm" + " " + "| grep 'k8s\\|haproxy'"
+		framework.Logf("Fetch vm details: %s ", vms)
+		vMsResult, err := sshExec(sshClientConfig, masterIp, vms)
+		if err != nil && vMsResult.Code != 0 {
+			fssh.LogResult(vMsResult)
+			return nil, fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				vms, masterIp, err)
+		}
+		if vMsResult.Stdout != "" {
+			vmListTemp := strings.Split(vMsResult.Stdout, "\n")
+			vmList = append(vmList, vmListTemp...)
+		}
+		for i := 0; i < len(vmList); i++ {
+			if vmList[i] != "" {
+				vmsList = append(vmsList, vmList[i])
+			}
+		}
+		vmList = nil
+	}
+	return vmsList, nil
+}
+
+// getDatastoreDetails method is used to fetch datastore details
+func getDatastoreDetails(vcAddress string, vCenterPort string, sshClientConfig *ssh.ClientConfig,
+	masterIp string, dataCenter []string) ([]string, error) {
+	var dsList []string
+	var datastores []string
+	// get vm details
+	ginkgo.By("Executing command for fetching datastore details")
+	for i := 0; i < len(dataCenter); i++ {
+		ds := govcLoginCmd(vcAddress, vCenterPort) + "govc ls /" + dataCenter[i] + "/datastore"
+		framework.Logf("Fetch datastore details: %s ", ds)
+		dsResult, err := sshExec(sshClientConfig, masterIp, ds)
+		if err != nil && dsResult.Code != 0 {
+			fssh.LogResult(dsResult)
+			return nil, fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				ds, masterIp, err)
+		}
+		if dsResult.Stdout != "" {
+			dsListTemp := strings.Split(dsResult.Stdout, "\n")
+			dsList = append(dsList, dsListTemp...)
+		}
+		for i := 0; i < len(dsList); i++ {
+			if dsList[i] != "" {
+				datastores = append(datastores, dsList[i])
+			}
+		}
+		dsList = nil
+	}
+	return datastores, nil
+}
+
+// setDataCenterLevelPermission is used to set data center level permissions for test user
+func setDataCenterLevelPermission(vcAddress string, vCenterPort string, sshClientConfig *ssh.ClientConfig,
```

Comment:
In all the `set*permissions()`, routines please accept two params,
* array of roles that you want to assign for the user
* whether to propagate the permissions or not